### PR TITLE
[ldid] Build against OpenSSL v3

### PR DESCRIPTION
**Pull request by staticfloat**

### CODE REVIEW

File: L/ldid/build_tarballs.jl [hunk 1]
```diff
@@ -30,7 +30,7 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("libplist_jll"),
-    Dependency("OpenSSL_jll"; compat="1.1.13"),
+    Dependency("OpenSSL_jll"; compat="3"),
```
Comment (giordano) [inline L/ldid/build_tarballs.jl:33]:
```suggestion
    Dependency("OpenSSL_jll"; compat="3.0.8"),
```

File: L/ldid/build_tarballs.jl [hunk 2]
```diff
@@ -3,7 +3,7 @@
 using BinaryBuilder
 
 name = "ldid"
-version = v"2.1.3" # <-- Fake version to build for new platforms
+version = v"2.1.4" # <-- Fake version to build with new OpenSSL compat bounds
```
Comment (giordano) [inline L/ldid/build_tarballs.jl:6]: The upstream repo has newer versions: https://git.saurik.com/ldid.git. BTW, why do we fork it?

Comment (staticfloat) [reply L/ldid/build_tarballs.jl:6]: It’s actually forked from https://github.com/tpoechtrager/ldid which made a bunch of build system changes

File: L/ldid/build_tarballs.jl [hunk 3]
```diff
@@ -30,7 +30,7 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("libplist_jll"),
-    Dependency("OpenSSL_jll"; compat="1.1.13"),
+    Dependency("OpenSSL_jll"; compat="3.0.8"),
```
Comment (giordano) [inline L/ldid/build_tarballs.jl:33]: Ok, we need a newer version for aarch64-freebsd:
```suggestion
    Dependency("OpenSSL_jll"; compat="3.0.15"),
```

File: L/ldid/build_tarballs.jl (hunk 2 above)
Comment (staticfloat) [reply L/ldid/build_tarballs.jl:6]: To actually answer your question, the `tpoechtrager` fork doesn't have any new commits, and glancing over the changes made in Saurik's source, I'm not sure it's worth my time to try and integrate them.  They don't seem that applicable to us.

**Pull request merged**